puppet/lib/puppet/util/puppetdb/http.rb

```diff
@@ -131,17 +131,18 @@ def self.failover_action(path_suffix, server_urls, sticky, ssl_context = nil, ht
 
       for server_url_index in server_try_order
         server_url = server_urls[server_url_index]
-        route = concat_url_snippets(server_url.request_uri, path_suffix)
+        route = concat_url_snippets(server_url.to_s, path_suffix)
 
         request_exception = with_http_error_logging(server_url, route) {
-          if ssl_context.nil?
-            http = Puppet::Network::HttpPool.http_instance(server_url.host, server_url.port)
-          else
-            http = Puppet::Network::HttpPool.connection(server_url.host, server_url.port, ssl_context: ssl_context)
-          end
+          # TODO: ssl_context?
```

Comment:
For the default case where you just want to use the existing set of certs, you don't need to pass an ssl_context. The client will take care of that. But if you had a specific requirement to use a particular set of certs, then you could build a context and pass it to the `Puppet::HTTP::Client#get`, etc methods.